### ISSUE TITLE
RStudio crashing on running highcharter treemap

### ISSUE THREAD

**Issue by chaudharyparth** [jbkunst/highcharter#319]
I'm creating TermDocumentMatrix in R and creating treemap from the sorted matrix. But as soon as i run `hc`, my RStudio crashes.

Code : 

```{r}
v <- getTermMatrix()
df <- data.frame(name = names(v), value = v, stringsAsFactors = FALSE)
rownames(df) <- NULL
#df <- setNames(df, c("name", "value"))
ds <- list_parse(df)

hc <- highchart() %>% 
hc_title(text = "A simple Treemap") %>% 
hc_add_series(data = ds, type = "treemap", colorByPoint = TRUE) %>%
hc_tooltip(crosshairs = TRUE, shared = TRUE, borderWidth = 5)
hc
```

`sessionInfo()`

```{r}
R version 3.1.2 (2014-10-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base
```

**Comment by jbkunst** [comment]
Hi @chaudharyparth 

...I don't know, I run your code without problems.

Joke! I can't see what it's going on if I can reproduce the problem. If you give me an example I can try to fix the problem.

;)

**Comment by wave-electron** [comment]
@chaudharyparth

Were you attempting to do something similar to this?

![screen shot 2017-06-05 at 3 06 23 pm](https://cloud.githubusercontent.com/assets/16273780/26771281/99ce1fa4-4a00-11e7-9d22-b0d3ba0d9823.png)

using this shiny example on github [Here](https://gist.github.com/jcheng5/9496297)

FYI. Adding your snippet of code into server.R produced the above screenshot of the Wordcloud shiny app. It didn't crash for me in Rstudio but was a bit laggy.

```
# Text of the books downloaded from:
# A Mid Summer Night's Dream:
#  http://www.gutenberg.org/cache/epub/2242/pg2242.txt
# The Merchant of Venice:
#  http://www.gutenberg.org/cache/epub/2243/pg2243.txt
# Romeo and Juliet:
#  http://www.gutenberg.org/cache/epub/1112/pg1112.txt
library(highcharter)

function(input, output, session) {
  observe({
  v <- getTermMatrix(input$selection)
  df <- data.frame(name = names(v), value = v, stringsAsFactors = FALSE)
  rownames(df) <- NULL
  #df <- setNames(df, c("name", "value"))
  ds <- list_parse(df)
  
  output$plot2 <-renderHighchart({ 
  
  hc <- highchart() %>% 
    hc_title(text = "A simple Treemap") %>% 
    hc_add_series(data = ds, type = "treemap", colorByPoint = TRUE) %>%
    hc_tooltip(crosshairs = TRUE, shared = TRUE, borderWidth = 5)
  hc
  
  })
  
  })
  # Define a reactive expression for the document term matrix
  terms <- reactive({
    # Change when the "update" button is pressed...
    input$update
    # ...but not for anything else
    isolate({
      withProgress({
        setProgress(message = "Processing corpus...")
        getTermMatrix(input$selection)
      })
    })
  })
  
  # Make the wordcloud drawing predictable during a session
  wordcloud_rep <- repeatable(wordcloud)
  
  output$plot <- renderPlot({
    v <- terms()
    wordcloud_rep(names(v), v, scale=c(4,0.5),
                  min.freq = input$freq, max.words=input$max,
                  colors=brewer.pal(8, "Dark2"))
  })
}




```
ui.R change to this

```
library(highcharter)
fluidPage(
  # Application title
  titlePanel("Word Cloud"),
  
  sidebarLayout(
    # Sidebar with a slider and selection inputs
    sidebarPanel(
      selectInput("selection", "Choose a book:",
                  choices = books),
      actionButton("update", "Change"),
      hr(),
      sliderInput("freq",
                  "Minimum Frequency:",
                  min = 1,  max = 50, value = 15),
      sliderInput("max",
                  "Maximum Number of Words:",
                  min = 1,  max = 300,  value = 100),
# Show Word Cloud      
plotOutput("plot")
    ),
    
    # Show treemap
    mainPanel(
      highchartOutput("plot2",height = "600px")
    )
  )
)


```

**Comment by chaudharyparth** [comment]
@wave-electron Yeah something like that but with a large corpus containing more than 1200 rows of data. 
@jbkunst Follow example of @wave-electron but with a larger dataset. 
 Other plots of `highcharter` library have no issues, only `treemap` hangs up my RStudio.

**Comment by wave-electron** [comment]
I loaded a much larger corpus containing 5756 rows using the the shiny app code above
`nrow(df) 
[1] 5756`

The file size was 99KB as txt.gz filetype. The ds list was 2.6 Mb in size. If you look at the tree map snapshot below its pushing the limits of the treemaps readability visually. It took about 25 seconds to load.

```
object.size(ds)
2737312 bytes
```

![screen shot 2017-06-06 at 9 28 27 am](https://cloud.githubusercontent.com/assets/16273780/26807615/ce85386e-4a9a-11e7-8937-c5f53a67881b.png)

When i created the treemap library it took even longer to process. so no improvement using that approach even though the object size is smaller. Took 1min 40 seconds to process. 


```
tm <- treemap(df, index = "name", vSize = "value", vColor = NULL, type = "index", palette = viridis(6))

object.size(tm)
784472 bytes
```
![screen shot 2017-06-06 at 9 45 55 am](https://cloud.githubusercontent.com/assets/16273780/26808171/711cd0ac-4a9e-11e7-8763-4cf254606570.png)

**Comment by chaudharyparth** [comment]
Sorry, to bother you guyz but this is a reproducible example i can come up with to demonstrate my `RStudio` crash.

```{r}
# url for data = "https://data.gov.in/resources/indian-railways-time-table-trains-available-reservation-03082015/download"

#steps : download data locally, then read the file

df<-read.csv("data/isl_wise_train_detail_03082015_v1.csv", stringsAsFactors = FALSE)
myCorpus <- Corpus(VectorSource(df))
myCorpus <- tm_map(myCorpus, content_transformer(tolower))
myCorpus <- tm_map(myCorpus, removeNumbers)
gramTokenizer <- function(x) NGramTokenizer(x, Weka_control(min = 1, max = 5))
myCorpus <- tm_map(myCorpus, PlainTextDocument)
myDTM <- TermDocumentMatrix(myCorpus, control = list(tokenize = gramTokenizer))
m <- as.matrix(myDTM)
m <- sort(rowSums(m), decreasing = TRUE)
df <- data.frame(name = names(m), value = m, stringsAsFactors = FALSE)
rownames(df) <- NULL
ds <- list_parse(df)
hc <- highchart() %>% 
	hc_title(text = "Treemap") %>% 
	hc_add_series(data = ds, type = "treemap", colorByPoint = TRUE) %>%
	hc_tooltip(crosshairs = TRUE, shared = TRUE, borderWidth = 5)
hc
```
Additional info : 
```{r}
> object.size(hc)
7451200 bytes
```

Also, my `sessionInfo()` is
```{r}
> sessionInfo()
R version 3.1.2 (2014-10-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base 
```

**Comment by wave-electron** [comment]
@chaudharyparth 

I ran your code and my rstudio session didn't crash, it was a bit slow creating the treemap... at least a minute or more rendering.

However, I did get the following error while originally running your code.

```
Error in simple_triplet_matrix(i, j, v, nrow = length(terms), ncol = length(corpus),  : 
  'i, j' invalid
```
 To fix the problem... I added  `myCorpus <- Corpus(VectorSource(myCorpus))` after the offending line of code  `myCorpus <- tm_map(myCorpus, PlainTextDocument)  & it fixed it.
```
sessionInfo()               
 version  R version 3.3.2 (2016-10-31)
 system   x86_64, darwin13.4.0        
 RStudio (1.0.136)           
```
